fix flake integration test endpoint slice mirroring updates

aojea opened the pull request:
/kind flake
```release-note
NONE
```

It can be reproduced if it runs with the same test in the file, it seems to be affected by the performance, if you add println statemens is harder to reproducer.

The problem seems to be that the test checks that all the slices have the correct field, but somehow, there are several slices present, and one of them doesn't have the corresponding fields

```
TestEndpointSliceMirroringUpdates/Update_annotations_but_triggertime (1.02s)
        endpointslicemirroring_test.go:402: Expected EndpointSlice to mirror annotations, expected map[endpoints.kubernetes.io/last-change-trigger-time:date foo2:bar2] received map[]
        endpointslicemirroring_test.go:360: Slice greater than 1 &EndpointSliceList{ListMeta:{ 1761  <nil>},Items:[]EndpointSlice{EndpointSlice{ObjectMeta:{test-updates-123-2qcms test-updates-123- test-endpointslice-mirroring-updates-2  247e600f-4e31-40ff-b02f-251ae7bc5720 1761 1 2021-03-09 21:26:31 +0100 CET <nil> <nil> map[endpointslice.kubernetes.io/managed-by:endpointslicemirroring-controller.k8s.io kubernetes.io/service-name:test-updates-123] map[foo2:bar2] [{v1 Endpoints test-updates-123 eeff15d9-8385-468e-8566-92afcb7cc146 0xc0295d707f 0xc0295d70a0}] []  []},Endpoints:[]Endpoint{Endpoint{Addresses:[10.0.0.1],Conditions:EndpointConditions{Ready:*true,Serving:nil,Terminating:nil,},Hostname:nil,TargetRef:nil,DeprecatedTopology:map[string]string{},NodeName:nil,Zone:nil,Hints:nil,},},Ports:[]EndpointPort{EndpointPort{Name:*,Protocol:*TCP,Port:*80,AppProtocol:nil,},},AddressType:IPv4,},EndpointSlice{ObjectMeta:{test-updates-123-br4jq test-updates-123- test-endpointslice-mirroring-updates-2  6841dbb4-fa9e-45bc-ba84-a7ee02ce7308 1759 1 2021-03-09 21:26:31 +0100 CET <nil> <nil> map[endpointslice.kubernetes.io/managed-by:endpointslicemirroring-controller.k8s.io kubernetes.io/service-name:test-updates-123] map[] [{v1 Endpoints test-updates-123 eeff15d9-8385-468e-8566-92afcb7cc146 0xc0295d712b 0xc0295d712c}] []  []},Endpoints:[]Endpoint{Endpoint{Addresses:[10.0.0.1],Conditions:EndpointConditions{Ready:*true,Serving:nil,Terminating:nil,},Hostname:nil,TargetRef:nil,DeprecatedTopology:map[string]string{},NodeName:nil,Zone:nil,Hints:nil,},},Ports:[]EndpointPort{EndpointPort{Name:*,Protocol:*TCP,Port:*80,AppProtocol:nil,},},AddressType:IPv4,},},}
FAIL
```

liggitt commented:
each integration test gets its own prefix in etcd:

```
etcdOptions := options.NewEtcdOptions(storagebackend.NewDefaultConfig(uuid.New().String(), nil))
```

aojea commented:
I can reproduce doing this
```
go test -count 3 ./test/integration/endpointslice -run TestEndpointSliceMirroring*
```

aojea committed:
fix integration endpointslicemirroring

it can happen that there are multiple slices, and some of them
outdated, so we relax the test to check that at least one
has the corresponding fields mirrored.

aojea commented:
@robscott this is the "easy" option, if we consider that there can be outdated slices present sometimes
/assign @robscott

robscott commented on test/integration/endpointslice/endpointslicemirroring_test.go:
```diff
@@ -396,10 +396,11 @@ func TestEndpointSliceMirroringUpdates(t *testing.T) {
-						return false, nil
+						continue
 					}
+					return true, nil
```
Mind adding the following comment here?

```
// This is a temporary workaround for https://github.com/kubernetes/kubernetes/issues/100033.
// TODO(robscott): When that issue is fixed, update this test to expect all EndpointSlices to meet this criteria.
```

aojea replied:
done

robscott commented:
Thanks @aojea! 

/lgtm

robscott commented:
Maybe @liggitt or @MrHohn can approve this?

robscott commented:
/assign @MrHohn

MrHohn reviewed (approved):
/approve

aojea commented:
/test pull-kubernetes-integration

these tests passed, failed some named as preemption
